File: c/media_geral_e_linha_matriz.c
//
// Autor: Maria Luiza Fernandes Prestes Cesar (RA: 832374)
//
// Data de Criação:     07/08/24
// Data de Modificação: 07/08/24
//
// Objetivo: encontrar média geral e média da linha em uma matriz
//

#include <stdio.h>

int main() {
    // Valores de input
    int linhas = 2;
    int colunas = 2;
    int matriz[2][2] = {{1, 2}, {3, 4}};

    // Valores iniciais
    int elementos_total = linhas * colunas;
    int soma_total = 0;

    // Somatória de cada elemento
    for (int i = 0; i < linhas; i++) {
        for (int j = 0; j < colunas; j++) {
            int element = matriz[i][j];
            soma_total += element;
        }
    }

    // Cálculo final
    double mediaGeral = soma_total / elementos_total;
    double mediaLinhas = soma_total / linhas;

    // Exibir valores finais
    printf("Media por elemento: %lf", mediaGeral);
    printf("Media por linha: %lf", mediaLinhas);
}

File: python/media_geral_e_linha_matriz.py
"""
 Autor: Maria Luiza Fernandes Prestes Cesar (RA: 832374)

 Data de Criação:     07/08/24
 Data de Modificação: 07/08/24

 Objetivo: calcula a média geral e média das linhas de uma matriz
"""


def main():
  linhas = 2
  colunas = 2
  matriz = [[1, 2], [3, 4]]

  elementos_total = linhas * colunas
  soma_total = 0

  for line in range(0, linhas):
    for column in range(0, colunas):
      element = matriz[line][column]
      soma_total += element

  media_geral = soma_total / elementos_total
  media_linhas = soma_total / linhas

  print(media_geral)
  print(media_linhas)


if __name__ == "__main__":
  main()